Update design for Spacewalk Issue and Redeeem

gonzamontiel opened the pull request:
### What's changed

- [x] Refactor: extract From input component (with asset selector) to be reusable 
- [x] Refactor: extract Issue and Redeem components (Feebox, Confirmation dialogs)
- [x] Add Yup schema validation to bridge forms
- [x] Import new icons (bell, settings, etc)
- [x] Change some of the default colors (Tolopea)
- [x]  Apply new Pendulum design for Issue (closes #221)
- [x]  Apply new Pendulum design for Redeem (closes #222)
- [x]  Apply new Amplitude design for Issue  (closes #247)
- [x]  Apply new Amplitude design for Redeem (closes #246)
- [x] Make Settings and vault selection also visible for Redeem (closes #249)
- [x] Add Settings Dialog box

ebma commented on src/pages/bridge/Issue/IssueValidationSchema.ts:
```diff
@@ -0,0 +1,13 @@
+    amount: Yup.number()
+      .typeError('Value is invalid.')
+      .transform(transformNumber)
+      .min(10, 'The minimum amount to bridge is 10.')
```
Is this a requirement by @pendulum-chain/product or where does it come from?

gonzamontiel replied:
I am not sure where I took that number from. @prayagd we want to set a minimum for bridging ?

prayagd replied:
@gonzamontiel i dont think so there is any minimum for bridging

ebma commented on src/pages/bridge/Redeem/RedeemValidationSchema.ts:
```diff
@@ -0,0 +1,31 @@
+    amount: Yup.number()
+      .typeError('Value is invalid.')
+      .transform(transformNumber)
+      .min(10, 'The specified amount is below the minimum (10).')
```
Same here.

gonzamontiel replied:
cc @prayagd

prayagd replied:
No minimum for bridging

ebma reviewed (changes requested):
The changes look great overall. However, I have some comments:
- The ‘max’ and ’50%’ buttons in the input form are confusing. We should change the order and maybe add an outline. Otherwise, it looks like it says ‘max 50%’ which is confusing. Also: Clicking on one of these buttons should not make me enter 50% or 100% of what is available to be issued by the vault, but 50% or 100% of what I can issue, ie. based on my available balance.
- I find it confusing that I can not toggle the expansion of the fee and disclaimer box but it has a click-away listener. I would expect that the box expands when I click on it _and stays expanded_ until I click on it again. With the current setup, it’s impossible for a user to see the detailed fee breakdown while entering an amount. They always have to click on the box again.
- We should change the disclaimer text but this should probably be provided by @pendulum-chain/product
-The size of the vault selection dialog is not responsive, which leads to an unfortunate scroll-bar when the vault selection dropdown is opened. I would recommend we either make the selector items hover over the dialog, or we make the dialog size adjust to the new size when the selection options are visible. Also, it would be nice if the dropdown selection would close immediately when I click on an item and not stay open until I click away.

gonzamontiel commented:
> The changes look great overall. However, I have some comments:
> 
> * The ‘max’ and ’50%’ buttons in the input form are confusing. We should change the order and maybe add an outline. Otherwise, it looks like it says ‘max 50%’ which is confusing. Also: Clicking on one of these buttons should not make me enter 50% or 100% of what is available to be issued by the vault, but 50% or 100% of what I can issue, ie. based on my available balance.

Okay, I didn't find them confusing because the hover only underlines the hovered one. Nevertheless, I can switch them to be `50%`  `MAX`, and add some spacing.
Regarding the values, is tricky, eh? I mean you have the max issuable tokens and max redeemable tokens, which are the values already shown in the current version. But it is true that this would be limited by the security deposit plus fees. Is that what you mean? So we would need to calculate it.

> * I find it confusing that I can not toggle the expansion of the fee and disclaimer box but it has a click-away listener. I would expect that the box expands when I click on it _and stays expanded_ until I click on it again. With the current setup, it’s impossible for a user to see the detailed fee breakdown while entering an amount. They always have to click on the box again.

It is SO annoying. Unfortunately this is the default behavior in Tailwind by design, and I don't really understand why they did it. I thought we could live with it, but what you say is true, the user won't be able to see the numbers change while typing. I'll find a way around it.

> * We should change the disclaimer text but this should probably be provided by @pendulum-chain/product
Still waiting  @prayagd ?

> * The size of the vault selection dialog is not responsive, which leads to an unfortunate scroll-bar when the vault selection dropdown is opened. I would recommend we either make the selector items hover over the dialog, or we make the dialog size adjust to the new size when the selection options are visible. Also, it would be nice if the dropdown selection would close immediately when I click on an item and not stay open until I click away.

Another one from Tailwind, they only close the dropwdown on blur. If you read the code I specifically added a blur() call on the item so it closes after selecting one item, and while this pattern is working fine in the Asset Selector, it's not working for the vault selector. I'll check.

gonzamontiel commented on src/components/Layout/links.tsx:
```diff
@@ -54,10 +54,10 @@ export const links: Links = ({ tenantName }) => [
     },
     prefix: <SpacewalkIcon />,
-    suffix: tenantName === TenantName.Pendulum ? <ComingSoonTag /> : <></>,
+    // suffix: tenantName === TenantName.Pendulum ? <ComingSoonTag /> : <></>,
```
```suggestion
    suffix: tenantName === TenantName.Pendulum ? <ComingSoonTag /> : <></>,
```

gonzamontiel commented on src/helpers/yup.ts:
```diff
@@ -1,6 +1,6 @@
-  if (!originalValue) return value;
-  if (typeof originalValue === 'string') value = Number(originalValue) ?? 0;
+  if (!originalValue) return 0;
+  if (typeof originalValue === 'string' && originalValue !== '') value = Number(originalValue) ?? 0;
```
Uhm, good point, it's redundant but it helps readability. As in, we convert the value to number only if it's a non empty string.

ebma commented:
> Regarding the values, is tricky, eh? I mean you have the max issuable tokens and max redeemable tokens, which are the values already shown in the current version. But it is true that this would be limited by the security deposit plus fees. Is that what you mean? So we would need to calculate it.

So about the `50%` and `MAX` button: while I said that we should take the available balance into account, this only works for the redeem/unbridging process and not for issuing. 
To elaborate, as a user I would expect that these buttons take 'my data' into account and don't make me choose values that are too high for my accounts and balances and which I cannot fulfill. _But_ we don't have information about the user's Stellar account balance when creating issue requests (because he can use any Stellar account), so I would propose we only show these buttons for the redeem process. Because for redeeming, we can just look at the user's Pendulum balance of the to-be-redeemed asset and base it on that. Does it make sense @gonzamontiel? Any opinion @prayagd?

gonzamontiel commented:
> > Regarding the values, is tricky, eh? I mean you have the max issuable tokens and max redeemable tokens, which are the values already shown in the current version. But it is true that this would be limited by the security deposit plus fees. Is that what you mean? So we would need to calculate it.
> 
> So about the `50%` and `MAX` button: while I said that we should take the available balance into account, this only works for the redeem/unbridging process and not for issuing. To elaborate, as a user I would expect that these buttons take 'my data' into account and don't make me choose values that are too high for my accounts and balances and which I cannot fulfill. _But_ we don't have information about the user's Stellar account balance when creating issue requests (because he can use any Stellar account), so I would propose we only show these buttons for the redeem process. Because for redeeming, we can just look at the user's Pendulum balance of the to-be-redeemed asset and base it on that. Does it make sense @gonzamontiel? Any opinion @prayagd?

Agreed in that the buttons usually refer to 'my data' as a user.
So when I first read your comment, I thought about answering this, like it only makes sense for Redeeming, but then the security deposit and transaction fee in the issuing process, are all taken from the Stellar amount ? meaning, I could make the transfer with 0 AMPE on my account? That's what I meant with my comment about calculating the max for me. If this is the case, it is not too bad UX to use the max button for the max issuable tokens, at the end, is the only max we could have.
In the redeem it does make sense to show the user's max balance. But this would leave us with a confusing UX for the Max buttons. So not sure.

gonzamontiel commented on src/config/apps/spacewalk.ts:
```diff
@@ -0,0 +1,24 @@
+        indexerUrl: string;
+        router: string;
+        oracle: string;
+        curve: string;
```
I deleted the spacewalk app because it was unused anyway.

gonzamontiel commented:
> > * I find it confusing that I can not toggle the expansion of the fee and disclaimer box but it has a click-away listener. I would expect that the box expands when I click on it _and stays expanded_ until I click on it again. With the current setup, it’s impossible for a user to see the detailed fee breakdown while entering an amount. They always have to click on the box again.
> 
> It is SO annoying. Unfortunately this is the default behavior in Tailwind by design, and I don't really understand why they did it. I thought we could live with it, but what you say is true, the user won't be able to see the numbers change while typing. I'll find a way around it.
> 

I fixed it. It is stil a bit weird because you can't copy values there, since clicking in te whole component will make the collapse to hide. I coudln't make it work easily just clicking on the title (the default behavior is hard to understand).

> > * The size of the vault selection dialog is not responsive, which leads to an unfortunate scroll-bar when the vault selection dropdown is opened. I would recommend we either make the selector items hover over the dialog, or we make the dialog size adjust to the new size when the selection options are visible. Also, it would be nice if the dropdown selection would close immediately when I click on an item and not stay open until I click away.
> 
> Another one from Tailwind, they only close the dropwdown on blur. If you read the code I specifically added a blur() call on the item so it closes after selecting one item, and while this pattern is working fine in the Asset Selector, it's not working for the vault selector. I'll check.

Fixed now.

prayagd commented:
> * We should change the disclaimer text but this should probably be provided by @pendulum-chain/product
Yes product would provide the text and will create a new ticket for this.

ebma commented:
> but then the security deposit and transaction fee in the issuing process, are all taken from the Stellar amount ? meaning, I could make the transfer with 0 AMPE on my account? That's what I meant with my comment about calculating the max for me. If this is the case, it is not too bad UX to use the max button for the max issuable tokens, at the end, is the only max we could have.

You don't have to deduce the fee for the '50%' and 'MAX" button of the input for the redeem amount. If you have a balance of 100 USDC on Pendulum, you should be able to put 100 USDC into the redeem form. The fees will eventually be deducted accordingly, so in the end you receive a little less but IMO this is the least confusing UX.

prayagd commented:
@gonzamontiel some more comments
1. "To amplitude" and "Back to stellar" look very closer. should can we add a bit of spacing between them or padding to the selections?
2.  I think so there is some confusion with Available and Max issueable, maybe not mentioned from my end in the ticket. But Available should be the wallet balance and max issueable/redeemable should be the capacity. So Available should only be shown in the Redeem section and not in the Issue section wrt to Max issueable can we show it the same way max redeemable is currently shown in the deployed version under the input field?
3. Currently the max redeemable shows 0.00 for all assets. Can we fix this?

gonzamontiel commented:
> @gonzamontiel some more comments
> 
> 1. "To amplitude" and "Back to stellar" look very closer. should can we add a bit of spacing between them or padding to the selections?

Fixed

> 2. I think so there is some confusion with Available and Max issueable, maybe not mentioned from my end in the ticket. But Available should be the wallet balance and max issueable/redeemable should be the capacity. So Available should only be shown in the Redeem section and not in the Issue section wrt to Max issueable can we show it the same way max redeemable is currently shown in the deployed version under the input field?

There's no design to show neither Max issuable and Available (I made that up actually). It would be nice to have the complete design, well thought by product. If that's missing then I made things up. I also made up how to show errors, I hope that's fine. I will try to put the max issuable somewhere.

> 3. Currently the max redeemable shows 0.00 for all assets. Can we fix this?

I will fix it.

gonzamontiel commented:
> You don't have to deduce the fee for the '50%' and 'MAX" button of the input for the redeem amount. If you have a balance of 100 USDC on Pendulum, you should be able to put 100 USDC into the redeem form. The fees will eventually be deducted accordingly, so in the end you receive a little less but IMO this is the least confusing UX.

I think there was a confusion re my comment. I know how the redeem works and that fees will be deducted from the original amount. I wasn't sure though, what happens when issuing. Balance is deducted from Stellar, we don't have control over it, I understand that. My comment was re the security deposit for issuing: will it be deducted from the converted XLM amount? Or I need to pay for that transaction from my own balance in AMPEs? 

![Screenshot from 2023-10-03 12 00 37](https://github.com/pendulum-chain/portal/assets/4713407/1938adaa-f965-4386-949f-aa939d29f03b)

ebma commented:
> My comment was re the security deposit for issuing: will it be deducted from the converted XLM amount? Or I need to pay for that transaction from my own balance in AMPEs?

The account that creates the issue request has to pay that deposit up-front with its own AMPE tokens.

prayagd commented:
> There's no design to show neither Max issuable and Available (I made that up actually). It would be nice to have the complete design, well thought by product. If that's missing then I made things up. I also made up how to show errors, I hope that's fine. I will try to put the max issuable somewhere.

That's absolutely fine and thanks for doing that, Product will try more that designs to be more complete. Otherwise all good from my end

gonzamontiel commented:
> > There's no design to show neither Max issuable and Available (I made that up actually). It would be nice to have the complete design, well thought by product. If that's missing then I made things up. I also made up how to show errors, I hope that's fine. I will try to put the max issuable somewhere.
> 
> That's absolutely fine and thanks for doing that, Product will try more that designs to be more complete. Also any reason we removed the max issuable from the design? Otherwise all good from my end

Sorry I will add it.

gonzamontiel commented:
> > My comment was re the security deposit for issuing: will it be deducted from the converted XLM amount? Or I need to pay for that transaction from my own balance in AMPEs?
> 
> The account that creates the issue request has to pay that deposit up-front with its own AMPE tokens.

Then we have a potential max.

ebma reviewed (commented):
Looks good overall. My only remark is about the shown errors, can we maybe improve that by putting the error related to the number input field near that component? The stellar field already shows the error as its label when needed.
It looks a bit out of place at the bottom. 

<img width="549" alt="image" src="https://github.com/pendulum-chain/portal/assets/6690623/489ad8fd-2c97-43ee-bb61-3c32fa8a5112">

gonzamontiel commented:
> Looks good overall. My only remark is about the shown errors, can we maybe improve that by putting the error related to the number input field near that component? The stellar field already shows the error as its label when needed. It looks a bit out of place at the bottom.
> 


Hey, okay I fixed it making it look like the input (so removing the `<Validation />` component). Now they look like this: 
![Screenshot from 2023-10-10 11 40 52](https://github.com/pendulum-chain/portal/assets/4713407/b51294e0-089f-4fbd-b761-00f906b925d7)
![Screenshot from 2023-10-10 11 40 28](https://github.com/pendulum-chain/portal/assets/4713407/542dc25a-787b-4d96-9db0-0a2c6533b528)

Disclaimer: the message now it has a final dot.

prayagd reviewed (approved):
Looks good to me, only can we make the error text bold?

gonzamontiel commented:
> Looks good to me, only can we make the error text bold?

I changed the designs, please approve if okay @ebma @prayagd 

![Screenshot from 2023-10-10 15 41 24](https://github.com/pendulum-chain/portal/assets/4713407/50e02dad-fa73-4225-9392-4656e2172426)
